### PR TITLE
Fix O125 code sinking dropping its paired deletion and breaking quoted values

### PR DESCRIPTION
**Pull request by bitwisecook**
## Summary

- `tcl opt` on the O125 KCS page's Before snippet emitted the sunk assignment twice, and with a quoted value the emitted text did not parse (`set msg "error; puts $msg`). Three defects sat behind that; the third surfaced once the first was fixed.
- **The paired deletion was dropped after selection.** The manager's resurrected-reference guard counts a def-elimination as resurrected when its variable still appears in a surviving replacement. The O125 insertion's replacement carries the sunk `set msg …` *and* the `$msg` that consumes it, so the guard read the deletion's own group-mate as a resurrection. A group-mate is now exempt from that scan, and a deletion the guard does drop takes the rest of its group with it, so a group stays all-or-nothing at this stage as it already did in overlap selection.
- **The quoted value lost its closing quote.** The pass replayed the assignment from the raw IR statement span, which follows the lexer's inner-end convention and stops *on* a quoted last word's closer. The segmenter deliberately does not widen quotes for `cmd.range`, so the truncation is by design at that layer and the consumer has to widen. Both the replayed text and the rewritten extents now go through `full_rewrite_span`, and the deletion takes the separator before the decision with it via `statement_delete_rewrite_range`, so the surviving text closes up instead of leaving a blank line.
- **The later-use scan missed a read spelled as a bare name.** It recognised only `$b` / `${b}`, which was masked while the deletion was being dropped. With the deletion surviving, `set b foo; if {$a} { puts $b }; puts [set b]` sank the assignment and left `b` undefined on the false path. `Statement::Call` already carries the `reads` and `defs` the lowerer resolved from the registry's `ArgRole::VarRead` / `VarWrite` positions, so the scan consults those rather than matching command names, and takes the name word of an `incr` as the read-modify-write it is. An argument holding a nested command substitution is not decomposed into statements, so `bareword_occurrences` covers its words. That scan and its `$var` sibling moved from `manager.rs` into `optimiser/helpers/var_refs.rs`, with their tests, so the two passes asking the question share one implementation.

Emitted source, before and after this change, for the page's snippet in both spellings:

| Input value | Before | After |
|---|---|---|
| `{error}` | `set msg {error}` kept, and copied into the branch | `if {$ok} { return } else { set msg {error}; log $msg }` |
| `"error"` | `set msg "error` copied into the branch, unterminated quote | `if {$ok} { return } else { set msg "error"; log $msg }` |

The shipped output matches the [O125 KCS page](docs/kcs/codes/kcs-optimisation-o125-code-sinking.md)'s After for both spellings, so that page is unchanged.

Also drops an unused parameter from `emit_sink`.

### Carried CI fix, unrelated to the O125 change

`test-ext` was red on the base branch, not on this diff: the extension suite runs 977 single-root tests where the manifest expects 976, because a test was added in e577f29 (#2022) without the pinned count moving with it. No branch carried a correction, so this PR ports it forward rather than sitting red; it no-ops if the same fix lands first. The count is pinned in four places by design, and all four move together: the manifest, both verifier scripts that re-check the manifest's own numbers, and the partition self-test. Detail in [this comment](https://github.com/bitwisecook/tcl-lsp/pull/2032#issuecomment-5603216018).

## Validation

- [x] I ran relevant tests/checks locally and listed the exact commands.

```
make rust-check                                  # exit 0
cargo test -p tcl-compiler                       # exit 0, 9517 tests across 66 binaries
bash scripts/dev/test-vscode-test-partitions.sh  # exit 0
```

New coverage:

- `rust/tcl-compiler/tests/optimiser.rs` — `code_sinking_o125_applies_both_grouped_edits_and_reparses` pins the exact emitted source for the braced and quoted values, asserts the assignment appears exactly once, and re-checks the result through the `tcl diag` surface for error-severity diagnostics. The quoted case raised `E202` before the fix.
- `rust/tcl-compiler/tests/optimiser.rs` — `code_sinking_o125_negatives` gains five by-name shapes: `info exists b`, `append b tail`, `incr b`, `puts [set b]`, and the same nested substitution inside a later branch body.
- `rust/tcl-compiler/src/optimiser/manager.rs` — two unit tests for the guard: the group-mate exemption, and a control proving an outside reference still drops the whole group.
- `rust/tcl-compiler/src/optimiser/helpers/var_refs.rs` — the moved scans keep their tests and gain word-boundary cases.

## Compiler / diagnostics checklist (when touching compiler behaviour, diagnostics, or analysis contracts)

- [x] Did this change alter a compiler fact contract? **No.** The inner-end convention and the group all-or-nothing rule are both unchanged; this makes the pass and the guard honour them.
- [x] If yes, I updated the owning design doc under `docs/design/compiler/` or `docs/design/contracts/`. **Updated anyway**, because `docs/design/compiler/o125-code-sinking.md` no longer described the code: the deletion extent and the span widening, the group all-or-nothing rule at both manager stages, what counts as a later use, and the test inventory.
- [x] If I added or changed a diagnostic or optimisation code, I updated its page under `docs/kcs/codes/` and linked it from `docs/kcs/codes/README.md`. No code added or changed; the O125 page already showed the intended result.
- [x] If I added a design doc, I linked it from `docs/design/README.md`. No new design doc.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01XBgwA1vhjnDtTrLdAo3vTi

### CODE REVIEW

File: rust/tcl-compiler/src/optimiser/manager.rs [hunk 1]
```diff
@@ -270,13 +279,26 @@ fn drop_def_elims_resurrected_by_replacements(
...
         .iter()
         .filter(|(i, var)| {
-            selected
-                .iter()
-                .enumerate()
-                .any(|(j, o)| j != *i && count_var_refs(&o.replacement, var) > 0)
+            let group = selected[*i].group;
+            selected.iter().enumerate().any(|(j, o)| {
+                j != *i
+                    && !(group.is_some() && o.group == group)
```
[the remark this answers is not in the document]
Comment (bitwisecook) [reply rust/tcl-compiler/src/optimiser/manager.rs:286]: Confirmed and fixed in 42a5c65.

Reproduced exactly as described. `set b foo; if {$a} { puts $b }; puts [set b]` sank the assignment into the branch, so on the false path `b` is undefined and the later read errors where the original printed nothing.

The fix is in the pass's own later-use scan rather than in the guard, so O125 is sound regardless of what the guard does. `Statement::Call` already carries the `reads` and `defs` the lowerer resolved from the registry's `ArgRole::VarRead` / `VarWrite` positions, so the scan consults those instead of matching command names, and takes the name word of an `incr` as the read-modify-write it is. An argument holding a nested command substitution is not decomposed into statements, so `bareword_occurrences` covers its words and makes `puts [set b]` a use as well.

That scan and its `$var` sibling moved out of `manager.rs` into `optimiser/helpers/var_refs.rs` with their tests, so the two passes asking the question share one implementation.

Five shapes are now asserted negatives in `code_sinking_o125_negatives`: `info exists b`, `append b tail`, `incr b`, `puts [set b]`, and the same nested substitution inside a later branch body. The stale note listing the bare-name case as a sound-but-spurious firing is gone, since it is neither now.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Comment by bitwisecook** [comment]
`test-ext` failed on 42a5c65, and the cause is not this PR's diff.

The three partitions all passed, and their aggregate is 976 passing plus 1 pending, so 977 completed. `editors/vscode/test-partitions.json` expects 976 completed and 975 passing, and the rollup's `verify-vscode-test-results.mjs` rejects the mismatch as `single-root aggregate expected outcome counts drifted`.

The extra test is `tcl-lsp.xcTranslate returns both output documents`, added to `editors/vscode/src/test/commandExecution.test.ts` in e577f29 (#2022). That commit did not update the expected count, so `test-ext` is red on the current base and on every branch that carries it, regardless of what the branch changes. This PR touches only `rust/tcl-compiler` and docs, none of which can add an extension test.

No fix for it existed on any branch, so I ported the count forward here rather than leaving the PR red. It no-ops for anyone who lands the same correction first. The count is pinned in four places on purpose, so a drift is loud, and all four move together:

- `editors/vscode/test-partitions.json`, the manifest itself
- `scripts/dev/verify-vscode-test-results.mjs`, which re-checks the manifest's own numbers before using them
- `scripts/dev/verify-vscode-test-partitions.mjs`, the same guard on the partition side
- `scripts/dev/test-vscode-test-partitions.sh`, the self-test, including the line it prints

`bash scripts/dev/test-vscode-test-partitions.sh` passes on the new values, and the latest `rust` is merged in. `make rust-check` and `cargo test -p tcl-compiler` are both green locally at 9517 tests.

---
_Generated by [Claude Code](https://claude.ai/code)_
